includes/admin/class-wcmypa-admin.php

```diff
@@ -971,12 +971,15 @@ private function printDeliveryDate(DeliveryOptions $deliveryOptions): void
 	 */
     private function getConfirmationData(DeliveryOptions $deliveryOptions): ?array
     {
-        $signatureTitle     = WCMP_Checkout::getDeliveryOptionsTitle(WCMYPA_Settings::SETTING_SIGNATURE_TITLE);
-        $onlyRecipientTitle = WCMP_Checkout::getDeliveryOptionsTitle(WCMYPA_Settings::SETTING_ONLY_RECIPIENT_TITLE);
-        $hasSignature       = $deliveryOptions->getShipmentOptions()->hasSignature();
-        $hasOnlyRecipient   = $deliveryOptions->getShipmentOptions()->hasOnlyRecipient();
+        $signatureTitle         = WCMP_Checkout::getDeliveryOptionsTitle(WCMYPA_Settings::SETTING_SIGNATURE_TITLE);
+        $onlyRecipientTitle     = WCMP_Checkout::getDeliveryOptionsTitle(WCMYPA_Settings::SETTING_ONLY_RECIPIENT_TITLE);
+        $hasSignature           = $deliveryOptions->getShipmentOptions()->hasSignature();
+        $hasOnlyRecipient       = $deliveryOptions->getShipmentOptions()->hasOnlyRecipient();
+        $deliveryOptionsEnabled = WCMYPA()->setting_collection->isEnabled(
+            WCMYPA_Settings::SETTING_DELIVERY_OPTIONS_ENABLED
+        );
 
-        if (! $deliveryOptions->getCarrier()) {
+        if (! $deliveryOptions->getCarrier() || ! $deliveryOptionsEnabled) {
```

Comment:
Ik zou `$deliveryOptionsEnabled` vooraan zetten in de if en alle variabelen die je nog niet nodig hebt onder de if statement zetten. Dat leest een stuk logischer weg

Reply:
done